From 60a97ee5b16a3300458611431c36427fa0dae9cd Mon Sep 17 00:00:00 2001
From: Murali Karicheri <m-karicheri2@ti.com>
Date: Wed, 4 Mar 2020 23:33:32 +0000
Subject: [PATCH 18/25] net: ti: icssg-prueth: use a single chn_name variable
 in prueth_start()

Use a single array variable for chn_name in prueth_start() by re-arranging
the code a bit.

Signed-off-by: Murali Karicheri <m-karicheri2@ti.com>
Acked-by: Roger Quadros <rogerq@ti.com>
---
 drivers/net/ti/icssg-prueth.c | 11 +++++------
 1 file changed, 5 insertions(+), 6 deletions(-)

diff --git a/drivers/net/ti/icssg-prueth.c b/drivers/net/ti/icssg-prueth.c
index 971a4631f1..19511638c9 100644
--- a/drivers/net/ti/icssg-prueth.c
+++ b/drivers/net/ti/icssg-prueth.c
@@ -279,22 +279,21 @@ static int prueth_start(struct udevice *dev)
 	struct prueth *priv = dev_get_priv(dev);
 	struct eth_pdata *pdata = dev->platdata;
 	int ret, i;
-	char tx_chn_name[16];
-	char rx_chn_name[16];
+	char chn_name[16];
 
 	icssg_class_set_mac_addr(priv->miig_rt, priv->slice,
 				 (u8 *)pdata->enetaddr);
 	icssg_class_default(priv->miig_rt, priv->slice);
 
 	/* To differentiate channels for SLICE0 vs SLICE1 */
-	snprintf(tx_chn_name, sizeof(tx_chn_name), "tx%d-0", priv->slice);
-	snprintf(rx_chn_name, sizeof(rx_chn_name), "rx%d", priv->slice);
+	snprintf(chn_name, sizeof(chn_name), "tx%d-0", priv->slice);
 
-	ret = dma_get_by_name(dev, tx_chn_name, &priv->dma_tx);
+	ret = dma_get_by_name(dev, chn_name, &priv->dma_tx);
 	if (ret)
 		dev_err(dev, "TX dma get failed %d\n", ret);
 
-	ret = dma_get_by_name(dev, rx_chn_name, &priv->dma_rx);
+	snprintf(chn_name, sizeof(chn_name), "rx%d", priv->slice);
+	ret = dma_get_by_name(dev, chn_name, &priv->dma_rx);
 	if (ret)
 		dev_err(dev, "RX dma get failed %d\n", ret);
 
-- 
2.17.1

